src/internal/io.js

```diff
@@ -104,16 +104,19 @@ export function spawn(fn, ...args) {
 
 const isForkedTask = task => task[TASK]
 
-export function join(task) {
-  if (is.array(task)) {
-    return task.map(join)
+export function join(...tasks) {
+  if (tasks.length > 1) {
```

Comment:
@yelouafi 
what do you think about those checks, it feela a little bit clunky, but I think this is what it takes to support both forms of the APIs, but maybe we should just drop one, so the code stays cleaner, aint sure however which one - `join(...tasks)` or `join([...tasks])`